app/scripts/modules/openstack/loadBalancer/configure/wizard/upsert.controller.js

```diff
@@ -0,0 +1,202 @@
+'use strict';
+
+let angular = require('angular');
+require('../../loadBalancer.less');
+
+module.exports = angular.module('spinnaker.loadBalancer.openstack.create.controller', [
+  require('angular-ui-router'),
+  require('../../../../core/loadBalancer/loadBalancer.write.service.js'),
+  require('../../../../core/loadBalancer/loadBalancer.read.service.js'),
+  require('../../../../core/account/account.service.js'),
+  require('../../../../core/modal/wizard/v2modalWizard.service.js'),
+  require('../../../../core/task/monitor/taskMonitorService.js'),
+  require('../../../../core/search/search.service.js'),
+  require('../../transformer.js'),
+  require('../../../../core/region/regionSelectField.directive.js'),
+  require('../../../subnet/subnetSelectField.directive.js'),
+  require('../../../floatingIp/floatingIpSelectField.directive.js'),
+])
+  .controller('openstackUpsertLoadBalancerController', function($scope, $uibModalInstance, $state,
+                                                                 application, loadBalancer, isNew, loadBalancerReader,
+                                                                 accountService, openstackLoadBalancerTransformer,
+                                                                 _, loadBalancerWriter, taskMonitorService) {
+    var ctrl = this;
+    $scope.isNew = isNew;
+
+    $scope.pages = {
+      location: require('./location.html'),
+      interface: require('./interface.html'),
+      destinations: require('./destinations.html'),
+      healthCheck: require('./healthCheck.html'),
+    };
+
+    $scope.state = {
+      accountsLoaded: false,
+      loadBalancerNamesLoaded: false,
+      submitting: false
+    };
+
+    $scope.regions = [];
+    $scope.subnetFilter = {};
+
+    $scope.protocols = ['HTTP', 'HTTPS'];
+    $scope.maxPort = 65535;
+    $scope.methods = [
+      { label: 'Round Robin', value: 'ROUND_ROBIN' },
```

Comment:
Honestly, I don't know, but I don't think so.  Rather, I think it is more likely that the number of options will grow slowly over time (dependent on the version of OpenStack in use). I don't see an API in https://wiki.openstack.org/wiki/Neutron/LBaaS/CLI that allows us to query it at run time.
This list is just a mirror of the current enumeration in the clouddriver code.